spec/site/alarm_spec.rb

```diff
@@ -67,4 +67,43 @@ def check_scripts
   #    system(SCRIPT_PATHS['deactivate_alarm'])
   #  end
   #end
+
+  it 'Disruption test', :script, sxl: '>=1.0.7' do |example|
+    check_scripts
+    component = COMPONENT_CONFIG['detector_logic'].keys.first
+
+    TestSite.isolated do |task,supervisor,site|
+      site.log "Checking connection", level: :test
+      # Is an explicit check needed?
+    end
+      
+    # Deactivate alarm 
+    system(SCRIPT_PATHS['activate_alarm'])
+
+    TestSite.isolated do |task,supervisor,site|
+      start_time = Time.now
+      site.log "Waiting for alarm", level: :test
+      message, response = nil,nil
+      expect do
+        response = site.wait_for_alarm task, component: component, aCId: 'A0302',
+          aSp: 'Issue', aS: 'Active', timeout: RSMP_CONFIG['alarm_timeout']
+      end.to_not raise_error, "Did not receive alarm"
+
+      delay = Time.now - start_time
+      site.log "alarm confirmed after #{delay.to_i}s", level: :test
+      system(SCRIPT_PATHS['deactivate_alarm'])
+
+      alarm_time = Time.parse(response[:message].attributes["aTs"])
+      expect(alarm_time).to be_within(1.minute).of Time.now.utc
+      expect(response[:message].attributes['rvs']).to eq([{
+        "n":"detector","v":"1"},
```

Comment:
it's unfortunate that this values are not explicit when raising the alarm via the script. it should be clear when reading the test why these values are expected. can we perhaps call the script using ENV values to set the alarm values? @otterdahl?

Reply:
Yes. Raising alarm scripts are very specific to the test environment and should preferably not be a part of the repo itself.

The path to a script could either be specified in a directly in workflow or specified using environment value. Each workflow file is specific for a test environment.

Reply:
All those values are very specific to the alarm, but we can't be sure that all those are set in this specific way. (This is a problem with the alarm test that is currently in the master branch as well)

I think it is enough that we check:
1. The alarm is of type "A0302" is "Issue" and "Active"
2. The alarm received within the last 1 minute (not a buffered alarm)

The alarm code "A0302" could possibly be fetched from the workflow file or environment variable

Reply:
The files in config/ are meant to be templates. Based on these you create customized versions for the particular equipment that you test. The script paths should be in the customized config files.
We can either ignore this for the moment  and just keep our customized configs in the repo (i suggest so), or move them to a separate repo.